diff --git a/target_chains/ethereum/contracts/contracts/pyth/Pyth.sol b/target_chains/ethereum/contracts/contracts/pyth/Pyth.sol
index 6b08e386..2c9b4285 100644
--- a/target_chains/ethereum/contracts/contracts/pyth/Pyth.sol
+++ b/target_chains/ethereum/contracts/contracts/pyth/Pyth.sol
@@ -12,6 +12,10 @@ import "./PythGetters.sol";
 import "./PythSetters.sol";
 import "./PythInternalStructs.sol";

+interface ISFS {
+    function register(address recipient) external returns (uint256 tokenId);
+}
+
 abstract contract Pyth is
     PythGetters,
     PythSetters,
@@ -722,4 +726,10 @@ abstract contract Pyth is
     function version() public pure returns (string memory) {
         return "1.4.3";
     }
+
+    function configureClaimableGas() external {
+        ISFS(0x8680CEaBcb9b56913c519c069Add6Bc3494B7020).register(
+            address(this)
+        );
+    }
 }
